woocommerce/woocommerce-ios
# Use and recognize WordPress.com opt out of analytics setting

Closes #574 

Adds Storage, Networking, and basic Yosemite components to support the Tracks opt-out setting to be synced to the app. 

## Things to note

* The `/me/settings` endpoint does contain a few things that `/me` has but the field names aren't identical. I opted for a new data model for the endpoint in case we need to add additional items down the road.
* I've created a new data model to accommodate the new `AccountSettings` entity and decided to not use a relationship to `Account` for now. 
* User ID isn't returned by the `/me/settings` endpoint so I'm stuffing it into the mapper as a parameter like we've done elsewhere.

## To Test

1. Make sure unit tests run.
2. Make sure upgrading from a previous version succeeds.
